src/background/messaging/rpc-methods/send-transfer.ts

```diff
@@ -1,36 +1,54 @@
 import { RpcErrorCode, SendTransferRequest } from '@btckit/types';
 
 import { RouteUrls } from '@shared/route-urls';
+import {
+  getRpcParamErrors,
+  rpcSendTransferValidator,
+  validateRpcParams,
+} from '@shared/rpc/methods/validation';
 import { makeRpcErrorResponse } from '@shared/rpc/rpc-methods';
+import { isDefined } from '@shared/utils';
 
 import {
+  RequestParams,
   getTabIdFromPort,
   listenForPopupClose,
   makeSearchParamsWithDefaults,
   triggerRequestWindowOpen,
 } from '../messaging-utils';
 
 export async function rpcSendTransfer(message: SendTransferRequest, port: chrome.runtime.Port) {
-  if (!message.params || !message.params.address || !message.params.amount) {
+  if (!message.params || !validateRpcParams(message.params, rpcSendTransferValidator)) {
```

Comment:
is the first check necessary, as `!validateRpcParams(undefined)` would also be false?

Reply:
It does need to be checked bc btckit let's the params be undefined. I separated it out into it's own error code thou.